internal/resource/nvml-mig-device.go

```diff
@@ -138,3 +138,11 @@ func (d nvmlMigDevice) GetPCIClass() (uint32, error) {
 	// GPU devices that support MIG do not support switching mode between graphics and compute, so they are always in compute mode.
 	return nvpci.PCI3dControllerClass, nil
 }
+
+func (d nvmlMigDevice) GetClusterUuid() (string, error) {
+	return "", fmt.Errorf("GetClusterUuid is not supported for MIG devices")
```

Comment:
```suggestion
	return "", fmt.Errorf("GetClusterUUID is not supported for MIG devices")
```

Reply:
Done

Reply:
What implications does returning an error here for applying the labels on the node? We should still apply them so long as all full GPUs have them and they match.

Reply:
yeah, it means we don't set a label if the current device is a MIG device. I was not sure if IMEX was supported on a MIG device. Can change to behave as a regular device

Reply:
No, it should not behave as a regular device because IMEX isn't supported on MIG.

Full GPUs should still be checked and if all of their clusterUUIDs and cliqueIDs match, then we should apply the label to the `nvidia.com/gpu` prefix as you are doing now. We should NOT apply it to `nvidia.com/mig-*` prefixes though, and we should also not apply it to `nvidia.com/gpu` in cases where there are no full GPUs on the node (which is possible both with mixed MIG mode or with single MIG mode strategies).

Reply:
yeah, the behavior of this specific section is not to set the label inside a MIG device, we will still label a node as expected. 
For example, a KIND cluster running each node on a MIG device won't get the labels on it's nodes.

Reply:
I still don't quite follow. If I have an 8 GPU machine with 4 nodes sliced into MIG devices, and 4 GPUs still being advertized as full GPUs, what in the current code is ensuring that the node still gets the `nvidia.com/gpu.clusterUUID`, `nvidia.com/gpu.cliqueID`, and `nvidia.com/gpu.imex-domain` labels?

Reply:
PR now on new direction